### ISSUE TITLE
IApplicationService.UpdateAsync corrupts ClientSecret

### ISSUE THREAD

**Issue by Khaos66** [thomasduft/openiddict-ui#19]
Not sure if it is meant to be used like this, but after updating an existing application the `ClientSecret` is changed.

```C#
var app = await _apps.GetAsync(appId);

var param = new ApplicationParameter()
{
   [...]
   ClientSecret = app.ClientSecret,
   [...]
}

await _apps.UpdateAsync(param);
```

Leaving the `ClientSecret` at its initial value of `null` will cause an exception, too.

So how am I supposed to update an existing application without changing the `ClientSecret`?

**Comment by thomasduft** [comment]
Hi @Khaos66 

That's correct! Up to now I have now other solution than deleting the existing record and always recreate a new one.

The reason is in the underlying OpenIddict-Core implementation and its functionality. It is encrypting the `ClientSecret` and only stores its hash which is correct IMO.

You can call it a bug in this repo but its not something to solve that easily other than come up with some data-flow / workflow around updating an application record that represents an non-interactive client (machine to machine communication).

**Comment by Khaos66** [comment]
Hi @thomasduft 
sorry I'm on vacation rn. I think I just used the OpenIDDict manager instance for updating the data without changing the `ClientSecret`. That worked fine, but I don't like it as I use your library as an abstraction layer above the OpenIDDict APIs and I didn't want to break that pattern...
Anyways, I get you're point and I'm fine with closing the issue.

**Comment by thomasduft** [comment]
Hi @Khaos66

Cool enjoy your holidays then...

Actually I just published v1.7.0 that makes updating confidential `Applications` more obvious (see description for [v1.7.0](https://github.com/thomasduft/openiddict-ui/releases/tag/v1.7.0))

Greets, Thomas
